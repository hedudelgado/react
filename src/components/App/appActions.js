import { HeaderInterface, InvalidSessionErrorCodeMap, InvalidSessionHelper } from 'mega-code';
import {
  getCountries,
  getNewsletterRegions,
  getNewsletterPreferences,
  getProfile,
  updateProfile,
  deleteCustomer,
} from '../../utils/api';

// actions
export const USER_LOGGED_IN_SUCCESSFULLY = 'USER_LOGGED_IN_SUCCESSFULLY';
export const COUNTRIES_RECEIVED = 'COUNTRIES_RECEIVED';
export const PROFILE_RECEIVED = 'PROFILE_RECEIVED';
export const REGIONS_RECEIVED = 'REGIONS_RECEIVED';
export const NEWSLETTER_RECEIVED = 'NEWSLETTER_RECEIVED';
export const SET_EDIT_VIEW = 'SET_EDIT_VIEW';
export const SET_NOTIFICATION_MESSAGE = 'SET_NOTIFICATION_MESSAGE';
export const PROFILE_UPDATED = 'PROFILE_UPDATED';
export const NEWSLETTER_UPDATED = 'NEWSLETTER_UPDATED';
export const RESET_NEWSLETTER_INFO = 'RESET_NEWSLETTER_INFO';
export const RESET_USER_INFO = 'RESET_USER_INFO';
// views
export const DEFAULT_VIEW = undefined;
export const USER_DETAILS_VIEW = 'USER_DETAILS_VIEW';
export const PASSWORD_VIEW = 'PASSWORD_VIEW';
export const PAYMENT_CARD_VIEW = 'PAYMENT_CARD_VIEW';
export const ADD_GUEST_VIEW = 'ADD_GUEST_VIEW';
export const REGULAR_GUESTS_VIEW = 'REGULAR_GUESTS_VIEW';
export const NEWSLETTER_VIEW = 'NEWSLETTER_VIEW';
export const ROOM_REQUIREMENTS_VIEW = 'ROOM_REQUIREMENTS_VIEW';
export const EXTRAS_PREFERENCES_VIEW = 'EXTRAS_PREFERENCES_VIEW';
export const MEMORABLE_WORD_VIEW = 'MEMORABLE_WORD_VIEW';
// errors
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const PASSWORD_ERROR = 'PASSWORD_ERROR';
export const NEWSLETTER_ERROR = 'NEWSLETTER_ERROR';
export const PAYMENT_CARD_ERROR = 'PAYMENT_CARD_ERROR';
export const USER_DETAILS_ERROR = 'USER_DETAILS_ERROR';
export const REGULAR_GUESTS_ERROR = 'REGULAR_GUESTS_ERROR';
export const MEMORABLE_WORD_ERROR = 'MEMORABLE_WORD_ERROR';
export const ROOM_REQUIREMENTS_ERROR = 'ROOM_REQUIREMENTS_ERROR';
export const EXTRAS_PREFERENCES_ERROR = 'EXTRAS_PREFERENCES_ERROR';

const checkForInvalidSession = (error, config, dispatch) => {
  if (InvalidSessionHelper
    .isSessionExpired(error, config)) {
    dispatch({
      type: RESET_USER_INFO,
    });
    dispatch({
      type: RESET_NEWSLETTER_INFO,
    });
    HeaderInterface.userLogOut(HeaderInterface.LOG_OUT_REASONS.SESSION_EXPIRED);
  }
};

export const getSettings = (email, sessionId, business) => (dispatch, getState) => {
  getProfile(email, sessionId, business).then((profile) => {
    dispatch({
      type: PROFILE_RECEIVED,
      profile,
    });
  }).catch((error) => {
    checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
    dispatch({
      type: PROFILE_ERROR,
      error,
    });
  });
  getCountries()
    .then((countries) => {
      dispatch({
        type: COUNTRIES_RECEIVED,
        countries,
      });
    });
  if (getState().application.showEmailPreferences) {
    getNewsletterRegions()
      .then((regions) => {
        dispatch({
          type: REGIONS_RECEIVED,
          regions,
        });
      });

    getNewsletterPreferences(email)
      .then((newsletter) => {
        dispatch({
          type: NEWSLETTER_RECEIVED,
          newsletter,
        });
      })
      .catch(e => dispatch({
        type: NEWSLETTER_ERROR,
        error: e,
      }));
  }
};

export const updateUserProfile = payload => dispatch => updateProfile(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
  return Promise.reject(error);
});

export const deleteUserProfile = payload => dispatch => deleteCustomer(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.DELETE_PROFILE.invalidSession, dispatch);
  return Promise.reject(error);
});

export default {
  getSettings,
};
