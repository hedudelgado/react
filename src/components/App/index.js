export { default } from './App';

export { default as AppActions } from './appActions';

export { default as AppContainer } from './AppContainer';

export { default as modalReducer } from './modalReducer';
