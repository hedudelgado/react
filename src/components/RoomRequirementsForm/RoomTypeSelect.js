import React from 'react';
import { array, bool, object } from 'prop-types';
import Select from '../../layout/Select';

const RoomTypeSelect = ({
  dictionary,
  type,
  hasChildren,
  hasMultipleAdults,
  ...fieldProps
}) => (
  <Select {...fieldProps}>
    {
      type
        .filter(({ code }) => (hasChildren ? code === 'FAM' : code !== 'FAM'))
        .filter(({ code }) => (hasMultipleAdults ? code !== 'SB' : true))
        .filter(({ code }) => (!hasMultipleAdults ? code !== 'TWIN' : true))
        .map(({ code, dictionaryKey }) => (
          <option
            key={code}
            value={code}
          >
            {dictionary[dictionaryKey]}
          </option>
        ))
    }
  </Select>
);

RoomTypeSelect.propTypes = {
  type: array.isRequired,
  hasChildren: bool,
  hasMultipleAdults: bool,
  dictionary: object.isRequired,
};

export default RoomTypeSelect;
