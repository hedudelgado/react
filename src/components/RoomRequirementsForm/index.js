export { default } from './RoomRequirementsFormContainer';

export { default as RoomTypeSelect } from './RoomRequirementsForm';
