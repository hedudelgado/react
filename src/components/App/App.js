import React, { Component } from 'react';
import {
  array,
  object,
  func,
  bool,
  node,
} from 'prop-types';
import {
  PASSWORD_VIEW,
  PAYMENT_CARD_VIEW,
  USER_DETAILS_VIEW,
  ROOM_REQUIREMENTS_VIEW,
  EXTRAS_PREFERENCES_VIEW,
} from './appActions';
import { ComponentTogglerContainer } from '../ComponentToggler';
import UserProfile from '../UserProfile';
import Password from '../Password';
import UserProfileFormContainer from '../UserProfileForm';
import PasswordFormContainer from '../PasswordForm';
import { PaymentCardContainer } from '../PaymentCard';
import { RegularGuestsContainer } from '../RegularGuests';
import { NewsletterContainer } from '../Newsletter';
import Modal from '../../layout/Modal';
import PaymentCardFormContainer from '../PaymentCardForm';
import { DeleteCustomerContainer } from '../DeleteCustomer';
import RoomRequirements from '../RoomRequirements';
import RoomRequirementsFormContainer from '../RoomRequirementsForm';
import ExtrasPreferences from '../ExtrasPreferences';
import ExtrasPreferencesFormContainer from '../ExtrasPreferencesForm';
import BaseStyle from '../../layout/BaseStyle';
import MemorableWord from '../MemorableWord';
import MoveAccount from '../MoveAccount';
import {
  STAYER, SELF, managePermissionFlag, USER_LOGGED_IN_EVENT,
} from '../../config.json';
import ManagePermission from '../ManagePermission';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleLoggedIn = this.handleLoggedIn.bind(this);
  }

  componentDidMount() {
    window.addEventListener('message', this.handleLoggedIn);
  }

  componentWillUnmount() {
    window.removeEventListener('message', this.handleLoggedIn);
  }

  async handleLoggedIn(event) {
    if (event.data.action === USER_LOGGED_IN_EVENT) {
      const { getSettings } = this.props;
      const { contactDetail: { email }, sessionId, business } = JSON.parse(event.data.profile);
      getSettings(email, sessionId, !!business);
    }
  }

  render() {
    const {
      countries,
      dictionary,
      profile,
      regions,
      newsletter,
      displayModal,
      handleModalClose,
      modalContent,
      userDataReceived,
      application: {
        showEmailPreferences,
      },
    } = this.props;

    const {
      business,
      business: {
        accessLevel = '',
        tetheredGuid = '',
      } = {},
    } = profile;

    const appReady = userDataReceived
      && (!showEmailPreferences || !!regions.length)
      && !!countries.length
      && (!showEmailPreferences || !!Object.keys(newsletter).length)
      && !!Object.keys(dictionary).length
      && !!Object.keys(profile).length;

    const showExtrasPreferences = !(accessLevel === SELF || accessLevel === STAYER);

    return (
      <div data-test="AccountSettings">
        {appReady
        && (
          <BaseStyle className="wb-typography app-container">
            <ComponentTogglerContainer
              viewName={USER_DETAILS_VIEW}
              view={UserProfile}
              edit={UserProfileFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PASSWORD_VIEW}
              view={Password}
              edit={PasswordFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PAYMENT_CARD_VIEW}
              view={PaymentCardContainer}
              edit={PaymentCardFormContainer}
            />
            {managePermissionFlag && (
              <ManagePermission dictionary={dictionary} />
            )}
            <RegularGuestsContainer />
            {showEmailPreferences && (
              <NewsletterContainer />
            )}
            <ComponentTogglerContainer
              viewName={ROOM_REQUIREMENTS_VIEW}
              view={RoomRequirements}
              edit={RoomRequirementsFormContainer}
            />
            {showExtrasPreferences && (
              <ComponentTogglerContainer
                viewName={EXTRAS_PREFERENCES_VIEW}
                view={ExtrasPreferences}
                edit={ExtrasPreferencesFormContainer}
              />
            )}
            {!!tetheredGuid && (
              <MemorableWord />
            )}
            {business && !tetheredGuid && (
              <MoveAccount />
            )}
            {!business && (
            <DeleteCustomerContainer />
            )}
            {displayModal && <Modal data-test="Modal" display content={modalContent} handleClose={handleModalClose} />}
          </BaseStyle>
        )
      }
      </div>
    );
  }
}

App.propTypes = {
  countries: array.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  getSettings: func.isRequired,
  displayModal: bool.isRequired,
  handleModalClose: func.isRequired,
  modalContent: node,
  newsletter: object.isRequired,
  regions: array.isRequired,
  userDataReceived: bool.isRequired,
  application: object.isRequired,
};

export default App;
