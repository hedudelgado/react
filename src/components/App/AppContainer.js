import { connect } from 'react-redux';
import { getSettings as getSettingsAction } from './appActions';
import { closeModal } from './modalActions';

import App from './App';

const mapStateToProps = ({
  modal: {
    display: displayModal,
    content: modalContent,
  },
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userProfile: {
    userDataReceived,
    profile,
  },
}) => ({
  profile,
  displayModal,
  modalContent,
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userDataReceived,
});

const mapDispatchToProps = dispatch => ({
  getSettings:
    (email, sessionId, business) => dispatch(getSettingsAction(email, sessionId, business)),
  handleModalClose: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
