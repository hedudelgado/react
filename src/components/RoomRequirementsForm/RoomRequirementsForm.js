import React from 'react';
import {
  object,
  func,
} from 'prop-types';
import Button from '../../layout/Button';
import Select from '../../layout/Select';
import { DEFAULT_VIEW } from '../App/appActions';
import RoomTypeSelect from './RoomTypeSelect';

const RoomRequirementsForm = ({
  touched,
  dictionary,
  setEditView,
  roomRequirements: {
    adults,
    children,
    type,
  },
  handleSubmit,
  handleBlur,
  handleChange,
  values,
  setFieldValue,
}) => {
  /**
   * @function handleFamilyRoom
   * @description Set field value to Family when there are more than 1 child.
   */
  const handleFamilyRoom = ({ target: { value } }) => {
    if (value > 0) {
      setFieldValue('type', 'FAM');
    }
    setFieldValue('children', value);
  };

  /**
   * @function hasMultipleAdults
   * @description Set field value to Double when there are multiple adults.
   */
  const hasMultipleAdults = ({ target: { value } }) => {
    if (value > 1) {
      setFieldValue('type', 'DB');
    }
    setFieldValue('adults', value);
  };

  return (
    <form
      data-test="room-requirements-form"
      name="room-requirements-form"
      onSubmit={handleSubmit}
    >
      <Select
        onBlur={handleBlur}
        onChange={hasMultipleAdults}
        defaultValue={values.adults}
        name="adults"
        id="adults"
        label={dictionary['roomrequirements.adults.label']}
        touched={touched.adults}
      >
        {
          adults.map(adult => (
            <option
              key={adult}
              value={adult}
            >
              {adult}
            </option>
          ))
        }
      </Select>
      <Select
        onBlur={handleBlur}
        onChange={handleFamilyRoom}
        defaultValue={values.children}
        name="children"
        id="children"
        label={dictionary['roomrequirements.children.label']}
        touched={touched.children}
      >
        {
          children.map(child => (
            <option
              key={child}
              value={child}
            >
              {child}
            </option>
          ))
        }
      </Select>
      <Select
        onBlur={handleBlur}
        onChange={handleChange}
        defaultValue={values.cotRequired}
        name="cotRequired"
        id="cotRequired"
        label={dictionary['roomrequirements.cot.label']}
        touched={touched.cotRequired}
      >
        <option key={dictionary['roomrequirements.cotrequired.true']} value="true">{dictionary['roomrequirements.cotrequired.true']}</option>
        <option key={dictionary['roomrequirements.cotrequired.false']} value="false">{dictionary['roomrequirements.cotrequired.false']}</option>
      </Select>
      <RoomTypeSelect
        type={type}
        dictionary={dictionary}
        hasChildren={parseInt(values.children, 10) > 0}
        hasMultipleAdults={parseInt(values.adults, 10) > 1}
        onBlur={handleBlur}
        onChange={handleChange}
        defaultValue={values.type}
        name="type"
        id="type"
        label={dictionary['roomrequirements.roomType.label']}
        touched={touched.type}
      />
      <Button
        className="wb-push--bottom"
        id="save-changed"
        color="primary"
        type="submit"
      >
        {dictionary['roomrequirements.button.save']}
      </Button>
      <Button
        data-test="closeForm"
        color="default"
        variant="text"
        className="underline block"
        fullWidth
        onClick={() => setEditView(DEFAULT_VIEW)}
      >
        {dictionary['roomrequirements.button.cancel']}
      </Button>
    </form>
  );
};

RoomRequirementsForm.propTypes = {
  touched: object.isRequired,
  handleBlur: func.isRequired,
  handleChange: func.isRequired,
  roomRequirements: object.isRequired,
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  values: object.isRequired,
  handleSubmit: func.isRequired,
  setFieldValue: func.isRequired,
};

export default RoomRequirementsForm;
