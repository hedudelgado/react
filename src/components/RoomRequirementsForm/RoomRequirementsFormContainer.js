import React, { PureComponent } from 'react';
import { func, object } from 'prop-types';
import { Formik } from 'formik';
import BaseStyle from '../../layout/BaseStyle';
import Typography from '../../layout/Typography';
import RoomRequirementsForm from './RoomRequirementsForm';
import {
  ROOM_REQUIREMENTS_VIEW,
  DEFAULT_VIEW,
  ROOM_REQUIREMENTS_ERROR,
} from '../App/appActions';
import Notification from '../../layout/Notification';
import { roomRequirements as roomRequirementsConfig } from '../../config.json';

export default class RoomRequirementsFormContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(values, actions) {
    const {
      setNotificationMessage,
      updateProfileStore,
      setEditView,
      updateProfile,
    } = this.props;
    const payload = this.createPayload(values);

    updateProfile(payload)
      .then(() => {
        setNotificationMessage(ROOM_REQUIREMENTS_VIEW);
        setEditView(DEFAULT_VIEW);
        actions.setSubmitting(false);
        updateProfileStore(payload);
      }).catch(() => {
        setNotificationMessage(ROOM_REQUIREMENTS_ERROR);
      });
  }

  createPayload(values) {
    const {
      adults,
      children,
      cotRequired,
      type,
    } = values;
    const { profile } = this.props;

    return {
      ...profile,
      bookingPreference: {
        ...profile.bookingPreference,
        roomRequirements: {
          adults: +adults,
          children: +children,
          cotRequired: cotRequired === 'true',
          type,
          hotelBrand: profile.bookingPreference
            && profile.bookingPreference.roomRequirements
            && profile.bookingPreference.roomRequirements.hotelBrand
            ? profile.bookingPreference.roomRequirements.hotelBrand
            : roomRequirementsConfig.defaults.hotelBrand,
        },
      },
    };
  }

  render() {
    const {
      profile,
      setEditView,
      dictionary,
      app: { notificationMessage },
    } = this.props;
    const {
      bookingPreference: {
        roomRequirements: {
          adults = roomRequirementsConfig.defaults.adults,
          children = roomRequirementsConfig.defaults.children,
          cotRequired = roomRequirementsConfig.defaults.cotRequired,
          type = roomRequirementsConfig.defaults.type,
        } = {},
      } = {},
    } = profile;

    return (
      <BaseStyle component="section" className="component-container">
        {notificationMessage === ROOM_REQUIREMENTS_ERROR && (
          <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
            {dictionary['generic.error']}
          </Notification>
        )}
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['roomrequirements.title.edit']}
        </Typography>

        <Formik
          data-test="room-requirements-form-wrapper"
          onSubmit={this.handleSubmit}
          initialValues={{
            adults,
            children,
            cotRequired,
            type,
          }}
          render={formikProps => (
            <RoomRequirementsForm
              {...formikProps}
              dictionary={dictionary}
              setEditView={setEditView}
              profile={profile}
              roomRequirements={roomRequirementsConfig}
            />
          )}
        />
      </BaseStyle>
    );
  }
}

RoomRequirementsFormContainer.propTypes = {
  app: object.isRequired,
  profile: object.isRequired,
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  setNotificationMessage: func.isRequired,
  updateProfileStore: func.isRequired,
  updateProfile: func.isRequired,
};
