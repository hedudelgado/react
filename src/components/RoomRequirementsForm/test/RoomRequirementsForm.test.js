import React from 'react';
import { shallow } from 'enzyme';
import RoomRequirementsForm from '../RoomRequirementsForm';
import { DEFAULT_VIEW } from '../../App/appActions';

describe('RoomRequirementsForm Component', () => {
  let wrapper;

  const props = {
    touched: {},
    dictionary: {
      'roomrequirements.title.edit': 'Edit room requirements',
      'roomrequirements.button.cancel': 'Cancel changes',
      'roomrequirements.button.save': 'Save changes',
      'roomrequirements.type.double': 'Double',
      'roomrequirements.type.single': 'Single',
      'roomrequirements.type.accessible': 'Accessible',
      'roomrequirements.type.family': 'Family',
    },
    roomRequirements: {
      adults: [1, 2],
      children: [0, 1, 2],
      type: [
        {
          dictionaryKey: 'roomrequirements.type.double',
          code: 'DB',
        },
        {
          dictionaryKey: 'roomrequirements.type.accessible',
          code: 'DIS',
        },
        {
          dictionaryKey: 'roomrequirements.type.family',
          code: 'FAM',
        },
        {
          dictionaryKey: 'roomrequirements.type.single',
          code: 'SB',
        },
      ],
    },
    profile: {
      bookingPreference: {
        roomRequirements: {
          type: 'DB',
          adults: 2,
          children: 0,
          cotRequired: false,
          hotelBrand: 'PI',
        },
      },
    },
    setEditView: jest.fn(),
    handleBlur: jest.fn(),
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    setFieldValue: jest.fn(),
    values: {
      adults: 0,
      children: 2,
      cotRequired: false,
      type: 'DB',
    },
  };

  beforeEach(() => {
    wrapper = shallow(<RoomRequirementsForm {...props} />);
  });

  describe('Renders RoomRequirementsForm Component Elements', () => {
    it('should render the component', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should go back to default view on Cancel', () => {
      wrapper.find('[data-test="closeForm"]').simulate('click');
      expect(props.setEditView).toHaveBeenCalledWith(DEFAULT_VIEW);
    });

    it('should submit the form', () => {
      wrapper.find('[name="room-requirements-form"]').simulate('submit');
      expect(props.handleSubmit).toHaveBeenCalledTimes(1);
    });

    it('should select Family room type if 1 or more children have been selected', () => {
      const event = { target: { value: 2 } };
      wrapper.find('[name="children"]').simulate('change', event);
      expect(wrapper.find('[name="type"]').dive().find('option[value="FAM"]')).toHaveLength(1);
      expect(wrapper.find('[name="type"]').dive().find('option[value="DB"]')).toHaveLength(0);
    });

    it('should not select Family room type if 0 children have been selected', () => {
      const newprops = {
        ...props,
        values: {
          adults: 0,
          children: 0,
          cotRequired: false,
          type: 'DB',
        },
      };

      wrapper = shallow(<RoomRequirementsForm {...newprops} />);

      const event = { target: { value: 0 } };
      wrapper.find('[name="children"]').simulate('change', event);
      expect(wrapper.find('[name="type"]').dive().find('option[value="FAM"]')).toHaveLength(0);
    });

    it('should select Double room type if multiple adults have been selected', () => {
      const newprops = {
        ...props,
        values: {
          adults: 2,
          children: 0,
          cotRequired: false,
          type: 'DB',
        },
      };

      wrapper = shallow(<RoomRequirementsForm {...newprops} />);

      const event = { target: { value: 2 } };
      wrapper.find('[name="adults"]').simulate('change', event);
      expect(wrapper.find('[name="type"]').dive().find('option[value="FAM"]')).toHaveLength(0);
      expect(wrapper.find('[name="type"]').dive().find('option[value="SB"]')).toHaveLength(0);
      expect(wrapper.find('[name="type"]').dive().find('option[value="DB"]')).toHaveLength(1);
    });

    it('should select Double room type if multiple adults have been selected but not allow Single', () => {
      const newprops = {
        ...props,
        values: {
          adults: 1,
          children: 0,
          cotRequired: false,
          type: 'DB',
        },
      };

      wrapper = shallow(<RoomRequirementsForm {...newprops} />);

      const event = { target: { value: 1 } };
      wrapper.find('[name="adults"]').simulate('change', event);
      expect(wrapper.find('[name="type"]').dive().find('option[value="FAM"]')).toHaveLength(0);
      expect(wrapper.find('[name="type"]').dive().find('option[value="SB"]')).toHaveLength(1);
      expect(wrapper.find('[name="type"]').dive().find('option[value="DB"]')).toHaveLength(1);
    });
  });
});
