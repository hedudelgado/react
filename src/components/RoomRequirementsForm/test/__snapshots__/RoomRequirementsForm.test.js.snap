// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomRequirementsForm Component Renders RoomRequirementsForm Component Elements should render the component 1`] = `
<form
  data-test="room-requirements-form"
  name="room-requirements-form"
  onSubmit={[MockFunction]}
>
  <Select
    defaultValue={0}
    halfWidth={false}
    hideIcon={false}
    hideMessage={false}
    id="adults"
    name="adults"
    onBlur={[MockFunction]}
    onChange={[Function]}
    placeholder=""
  >
    <option
      key="1"
      value={1}
    >
      1
    </option>
    <option
      key="2"
      value={2}
    >
      2
    </option>
  </Select>
  <Select
    defaultValue={2}
    halfWidth={false}
    hideIcon={false}
    hideMessage={false}
    id="children"
    name="children"
    onBlur={[MockFunction]}
    onChange={[Function]}
    placeholder=""
  >
    <option
      key="0"
      value={0}
    >
      0
    </option>
    <option
      key="1"
      value={1}
    >
      1
    </option>
    <option
      key="2"
      value={2}
    >
      2
    </option>
  </Select>
  <Select
    defaultValue={false}
    halfWidth={false}
    hideIcon={false}
    hideMessage={false}
    id="cotRequired"
    name="cotRequired"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder=""
  >
    <option
      value="true"
    />
    <option
      value="false"
    />
  </Select>
  <RoomTypeSelect
    defaultValue="DB"
    dictionary={
      Object {
        "roomrequirements.button.cancel": "Cancel changes",
        "roomrequirements.button.save": "Save changes",
        "roomrequirements.title.edit": "Edit room requirements",
        "roomrequirements.type.accessible": "Accessible",
        "roomrequirements.type.double": "Double",
        "roomrequirements.type.family": "Family",
        "roomrequirements.type.single": "Single",
      }
    }
    hasChildren={true}
    hasMultipleAdults={false}
    id="type"
    name="type"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    type={
      Array [
        Object {
          "code": "DB",
          "dictionaryKey": "roomrequirements.type.double",
        },
        Object {
          "code": "DIS",
          "dictionaryKey": "roomrequirements.type.accessible",
        },
        Object {
          "code": "FAM",
          "dictionaryKey": "roomrequirements.type.family",
        },
        Object {
          "code": "SB",
          "dictionaryKey": "roomrequirements.type.single",
        },
      ]
    }
  />
  <Button
    className="wb-push--bottom"
    color="primary"
    fullWidth={false}
    href={null}
    id="save-changed"
    onClick={[Function]}
    submit={false}
    type="submit"
    variant="button"
  >
    Save changes
  </Button>
  <Button
    className="underline block"
    color="default"
    data-test="closeForm"
    fullWidth={true}
    href={null}
    onClick={[Function]}
    submit={false}
    variant="text"
  >
    Cancel changes
  </Button>
</form>
`;
